src/Microsoft.Extensions.Primitives/CompositeChangeToken.cs

```
@@ -0,0 +1,126 @@
+        {
+            if (!_canBeChanged)
+            {
+                return NullDisposable.Singleton;
+            }
+
+            EnsureCallbacksRegistered();
+            return _cancellationTokenSource.Token.Register(callback, state);
+        }
+
+        /// <inheritdoc />
+        public bool HasChanged
+        {
+            get
+            {
+                for (var i = 0; i < ChangeTokens.Count; i++)
+                {
+                    if (ChangeTokens[i].HasChanged)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        /// <inheritdoc />
+        public bool ActiveChangeCallbacks
+        {
+            get
+            {
+                for (var i = 0; i < ChangeTokens.Count; i++)
+                {
+                    if (ChangeTokens[i].ActiveChangeCallbacks)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        private void EnsureCallbacksRegistered()
+        {
+            if (_registeredCallbackProxy)
+            {
+                return;
+            }
+
+            for (var i = 0; i < ChangeTokens.Count; i++)
+            {
+                if (ActiveChangeCallbacks)
+                {
+                    ChangeTokens[i].RegisterChangeCallback(_onChangeDelegate, this);
+                }
+            }
+            _registeredCallbackProxy = true;
+        }
+
+        private static void OnChange(object state)
+        {
+            var compositeChangeTokenState = (CompositeChangeToken)state;
+            if (!compositeChangeTokenState._canBeChanged)
+            {
+                return;
+            }
+
+            compositeChangeTokenState._canBeChanged = false;
+            compositeChangeTokenState._cancellationTokenSource.Cancel();
```

try/catch, swallow exceptions.